app/src/main/java/org/eu/exodus_privacy/exodusprivacy/manager/sync/SyncManager.kt

```diff
@@ -0,0 +1,215 @@
+
+                val exodusApp = ExodusApplication(
+                    packageName = app.packageName,
+                    name = app.name,
+                    icon = app.icon,
+                    versionName = app.versionName,
+                    versionCode = app.versionCode,
+                    permissions = app.permissions,
+                    exodusVersionName = latestApp.version_name,
+                    exodusVersionCode = latestApp.version_code.toLongOrZero(),
+                    exodusTrackers = latestApp.trackers,
+                    source = app.source,
+                    report = latestApp.report,
+                    created = latestApp.created,
+                    updated = latestApp.updated,
+                )
+
+                syncedApps.add(exodusApp)
+
+                onAppSync(latestApp)
+            }
+            syncedApps.forEach { exodusApp ->
+                databaseRepository.saveApp(exodusApp)
+            }
+        }
+    }
+
+    private suspend fun syncAllTrackers() = supervisorScope {
+        val list = try {
+            apiRepository.getAllTrackers()
+        } catch (e: Exception) {
+            e.propagateCancellation()
+            Log.e(TAG, "Unable to fetch trackers.", e)
+            null
+        }
+        list?.trackers?.forEach { (key, tracker) ->
+            launch {
+                ensureActive()
+                val trackerData = TrackerData(
+                    id = key.toInt(),
+                    categories = tracker.categories,
+                    code_signature = tracker.code_signature,
+                    creation_date = tracker.creation_date,
+                    description = tracker.description,
+                    name = tracker.name,
+                    network_signature = tracker.network_signature,
+                    website = tracker.website,
+                )
+                try {
+                    databaseRepository.saveTrackerData(trackerData)
+                } catch (e: IOException) {
+                    Log.e(TAG, "Unable to save trackers.", e)
+                }
+            }
+        }
+    }
+
+    private suspend fun updateTrackerDetails(
+        appDetails: List<AppDetails>,
+    ) {
+        supervisorScope {
+            appDetails.forEach { detail ->
+                launch {
+                    detail.trackers.forEach { trackerId ->
+                        setTrackerActive(trackerId, detail.handle)
+                    }
+                }
+            }
+            updateAppsWithTrackersCount(appDetails.size)
+        }
+    }
+
+    private suspend fun removeDeletedApps(
+        currentApps: List<Application>,
+    ) {
+        withContext(ioDispatcher) {
+            try {
+                val savedApps = databaseRepository.getAllPackageNames()
+                val currentPackageNames = currentApps.map { it.packageName }
+                val isAppListUpdated = currentPackageNames.size != savedApps.size ||
```

Comment:
Update app list if current database size is different compared to new size, or current database is empty, or current database and saved database are not equal